compiler/rustc_privacy/src/lib.rs

```diff
@@ -1298,7 +1298,17 @@ impl SearchInterfaceForPrivateItemsVisitor<'_> {
 
     fn ty(&mut self) -> &mut Self {
         self.in_primary_interface = true;
-        self.visit(self.tcx.type_of(self.item_def_id).instantiate_identity());
+        let ty = self.tcx.type_of(self.item_def_id).instantiate_identity();
+
+        // if `in_assoc_ty`, attempt to normalize `ty`
+        let maybe_normalized_ty = if self.in_assoc_ty {
```

Comment:
This visitor is supposed to normalize everywhere, not just in associated types.

If that's not possible for some technical reasons, then it's fine to normalize only in a subset of cases, just need to leave an explaining comment.

Reply:
Yes, removing the `self.in_assoc_ty` check caused quite a few unexpected errors (e.g., recursion limit errors). I'll leave a comment to that effect.

I don't quite have the expertise to debug these in short-order, and the narrow issue of normalization in associated types is what's blocking me from releasing zerocopy 0.8. If it's possible to merge this narrow fix before 1.80 branches from master, I'd be quite grateful.

Reply:
Since we're delaying it to 1.81 anyway, let's try out both using the new solver and always trying to normalize, even for non-assoc types.